pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -636,19 +766,23 @@ func (vs *VSphere) CreateVolume(volumeOptions *vclib.VolumeOptions) (volumePath
 		var datastore string
 		// Default datastore is the datastore in the vSphere config file that is used to initialize vSphere cloud provider.
 		if volumeOptions.Datastore == "" {
-			datastore = vs.cfg.Global.DeafultDatastore
+			datastore = vs.cfg.Global.DefaultDatastore
 		} else {
 			datastore = volumeOptions.Datastore
 		}
 		// Create context
 		ctx, cancel := context.WithCancel(context.Background())
 		defer cancel()
 		// Ensure client is logged in and session is valid
-		err = vs.conn.Connect(ctx)
+		vsi, err := vs.getVSphereInstance(vmNameToNodeName(vs.hostName))
+		if err != nil {
+			return "", err
+		}
+		err = vsi.conn.Connect(ctx)
 		if err != nil {
 			return "", err
 		}
-		dc, err := vclib.GetDatacenter(ctx, vs.conn, vs.cfg.Global.Datacenters)
+		dc, err := vclib.GetDatacenter(ctx, vsi.conn, vs.cfg.Global.Datacenter)
```

Comment:
This method will be updated once we implement CreateVolume for multi-VC. Ideally we would expect to just use vsi.dc here since node manager would have already discovered the location of the node. But that is not in the scope of this task.